packages/hooks/src/usePoller.ts

```diff
@@ -0,0 +1,74 @@
+import { useState, useRef, useEffect } from 'react';
+import useEventListener from './useEventListener';
+
+function useVisibilityChange() {
+  const isVisibilityStateVisible = () => document.visibilityState === 'visible';
```

Comment:
I dig this!

If we're doing this sort of fancy thing, what are your thoughts on slowing down polling on slower internet connections using the `NetworkInformation` API? (as an option for the hook, not a default of course)

Reply:
How much should we slow polling? I assume most consumers would set their own interval, so if we exposed a hook that returned the `NetworkInformation` data then they could set the interval depending on the network speed.

Reply:
Hmm... Yeah, I suppose exposing a hook for that might be a better path. I was thinking something like a `networkSpeedFactor` option that slows the polling by the passed multiple that could be less effort, though that would limit how granular a consumer could be.

Reply:
There's no guarantee that a poller is _necessarily_ making network requests, right? Probably best to allow the consumer to make that choice.

Reply:
Yeah, I 100% agree that if we were to even bake it into usePoller, it shouldn't be enabled by default.